src/swap/utils.ts

```diff
@@ -1,243 +1,250 @@
+  isSwapQuoteErrorCausedByAmount
+} from "./common/utils";
+import {getAssetId} from "../util/asset/assetUtils";
 import {isPoolEmpty} from "../util/pool/common";
-import OutputAmountExceedsAvailableLiquidityError from "../util/error/OutputAmountExceedsAvailableLiquidityError";
+import SwapQuoteError, {SwapQuoteErrorType} from "../util/error/SwapQuoteError";
 
 /**
- * Gets quotes for swap from each pool passed as an argument,
- * and returns the best quote (with the highest rate).
+ * Gets the best quote for swap from the pools and swap router and returns the best option.
  */
-export function getQuote(params: {
-  type: SwapType;
-  pools: {info: V1PoolInfo | V2PoolInfo; reserves: PoolReserves}[];
-  assetIn: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
-  assetOut: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
-  amount: number | bigint;
-}): Promise<SwapQuoteWithPool> {
-  if (params.pools.every((pool) => isPoolEmpty(pool.reserves))) {
-    throw new Error("No pools available for swap");
+export function getQuote(params: GetSwapQuoteParams): Promise<SwapQuote> {
+  const {type, pools, isSwapRouterEnabled} = params;
+
+  if (!isSwapRouterEnabled && pools.every((pool) => isPoolEmpty(pool.reserves))) {
+    throw new SwapQuoteError(
+      SwapQuoteErrorType.NoAvailablePoolError,
+      "There is not an available pool for this swap. However, you can enable swap router and try to perform this swap with multiple pools."
+    );
   }
 
-  if (params.type === SwapType.FixedInput) {
+  if (type === SwapType.FixedInput) {
     return getFixedInputSwapQuote(params);
+  } else if (type === SwapType.FixedOutput) {
+    return getFixedOutputSwapQuote(params);
   }
 
-  return getFixedOutputSwapQuote(params);
+  throw new SwapQuoteError(SwapQuoteErrorType.InvalidSwapTypeError, "Invalid swap type");
 }
 
 /**
  * Given a list of swap quotes, it filters out the invalid quotes.
  * Validity of the quote is checked by if the promise was successfully resolved
  * and an actual quote object is available on the response
  */
-function validateQuotes(
-  promises: Promise<SwapQuoteWithPool>[]
-): Promise<SwapQuoteWithPool[]> {
+function validateQuotes(promises: Promise<SwapQuote>[]): Promise<SwapQuote[]> {
   return Promise.allSettled(promises).then((results) => {
-    if (
-      results.every(
-        (result) =>
-          result.status === "rejected" &&
-          result.reason instanceof OutputAmountExceedsAvailableLiquidityError
-      )
-    ) {
-      throw new OutputAmountExceedsAvailableLiquidityError();
+    if (results.every((result) => result.status === "rejected")) {
+      const [v1_1PoolError, v2PoolError] = (results as PromiseRejectedResult[]).map(
+        (result) => result.reason
+      ) as SwapQuoteError[];
+
+      if (
+        isSwapQuoteErrorCausedByAmount(v1_1PoolError) &&
+        !isSwapQuoteErrorCausedByAmount(v2PoolError)
+      ) {
+        throw v1_1PoolError;
+      }
+
+      throw v2PoolError;
```

Comment:
if v2 pool doesn't exist and the router wasn't enabled, we should still display error that comes from v1 pool even if it isn't about liquidity. I guess in addition to the "type" field, it may be useful to add a "source" field to the SwapQuoteError and type it as a union of `"v1_1" | "v2" | "router"`